presto-main/src/main/java/com/facebook/presto/operator/aggregation/state/ReduceAggregationStateFactory.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation.state;
+
+import com.facebook.presto.array.ObjectBigArray;
+import com.facebook.presto.common.block.Block;
+import com.facebook.presto.spi.function.AccumulatorStateFactory;
+import io.airlift.slice.Slice;
+import org.openjdk.jol.info.ClassLayout;
+
+public class ReduceAggregationStateFactory
+        implements AccumulatorStateFactory<ReduceAggregationState>
+{
+    @Override
+    public ReduceAggregationState createSingleState()
+    {
+        return new SingleState();
+    }
+
+    @Override
+    public Class<? extends ReduceAggregationState> getSingleStateClass()
+    {
+        return SingleState.class;
+    }
+
+    @Override
+    public ReduceAggregationState createGroupedState()
+    {
+        return new GroupedState();
+    }
+
+    @Override
+    public Class<? extends ReduceAggregationState> getGroupedStateClass()
+    {
+        return GroupedState.class;
+    }
+
+    private static long getObjectSize(Object value)
+    {
+        if (value instanceof Block) {
+            return ((Block) value).getRetainedSizeInBytes();
+        }
+        else if (value instanceof Slice) {
+            return ((Slice) value).getRetainedSize();
+        }
+
+        return 0;
+    }
+
+    public static class GroupedState
+            extends AbstractGroupedAccumulatorState
+            implements ReduceAggregationState
+    {
+        private static final int INSTANCE_SIZE = ClassLayout.parseClass(GroupedState.class).instanceSize();
+        private final ObjectBigArray<Object> values = new ObjectBigArray<>();
```

Comment:
break a line between const and variables

Reply:
done